cmd/gateway/azure/gateway-azure.go

```diff
@@ -1202,19 +1211,18 @@ func (a *azureObjects) CompleteMultipartUpload(ctx context.Context, bucket, obje
 	if err != nil {
 		return objInfo, azureToObjectError(err, bucket, object)
 	}
-	if len(metadata.Metadata) > 0 {
-		objBlob.Metadata, objBlob.Properties, err = s3MetaToAzureProperties(ctx, metadata.Metadata)
-		if err != nil {
-			return objInfo, azureToObjectError(err, bucket, object)
-		}
-		err = objBlob.SetProperties(nil)
-		if err != nil {
-			return objInfo, azureToObjectError(err, bucket, object)
-		}
-		err = objBlob.SetMetadata(nil)
-		if err != nil {
-			return objInfo, azureToObjectError(err, bucket, object)
-		}
+	objBlob.Metadata, objBlob.Properties, err = s3MetaToAzureProperties(ctx, metadata.Metadata)
+	objBlob.Metadata["md5sum"] = fmt.Sprintf("%s-%d", hex.EncodeToString(hasher.Sum(nil)), len(uploadedParts))
```

Comment:
@cehoffman, you can define a link to getCompleteMultipartMD5 in gateway-common.go and reuse the function which computes MD5Sum

Reply:
Didn't even think to look for a helper doing the job already. Refactored to expose `getCompleteMultipartMD5` as `ComputeCompleteMultipartMD5` and use that here.